recipes/CommonVoice/self-supervised-learning/wav2vec2/train.py

```diff
@@ -123,6 +123,10 @@ def fit_batch(self, batch):
 
                 # anneal lr every update
                 self.hparams.noam_annealing(self.optimizer)
+                # decay gumbel temperature until it reaches the minimum value
+                self.modules.wav2vec2.module.model.set_gumbel_temperature(
+                    max(self.hparams.gumbel_temp_max * self.hparams.gumbel_temp_decay ** self.step //
+                        self.hparams.gradient_accumulation, self.hparams.gumbel_temp_min))
```

Comment:
It's not clear to me, will this work correctly for when training with 1 GPU and when training with multiple? In the huggingface code there is a check for this.

Reply:
The paper has this line: "The Gumbel softmax temperature tau is annealed from 2
to a minimum of 0.5 for BASE and 0.1 for LARGE by a factor of 0.999995 at every update.", but I can can checkout the fairseq implementation.  For my hack, I have the values hardcoded running on a single dgx across the 8 V100s with the data_parallel_backend flag (I didnt properly use the yaml file, but I can test tomorrow).  I suspect its uncommon to use that data_parallel_backend flag with the recipe since the recipe receives an array of loss values and erroneously attempts to solve the gradients.  I updated to take the mean prior.

Reply:
@thadeous any updates/comments?